src/Core.php

```diff
@@ -10,7 +10,7 @@ class Core implements CoreInterface
     public function random(GeneratorInterface $generator, $size, $alphabet = CoreInterface::SAFE_SYMBOLS)
     {
         $len = strlen($alphabet);
-        $mask = (2 << log($len - 1) / M_LN2) - 1;
+        $mask = (2 << (int) (log($len - 1) / M_LN2)) - 1;
```

Comment:
@matt-harvey Please take a look. <. _<.

Reply:
Thanks for the feedback @hidehalo . My aim in this PR was to fix the deprecation warning only, not to change the behaviour in any other way. AFAICS, the current behaviour on `master` branch, converting the float to an integer with implicit conversion, is to truncate the float i.e. always round it down. Adding an `(int)` to make the conversion explicit, doesn't change that behaviour; the only thing it does is to make the deprecation warning go away.

I'm not familiar enough with the algorithm here to know whether it would be better to round than to truncate here. It's your call really... Although, merging this PR wouldn't make the algorithm _less_ accurate; and then the rounding issue could be addressed separately, if desired. What do you think?

Reply:
NIce, thanks @hidehalo !